pages/index.js

```diff
@@ -17,151 +18,155 @@ const dayStartTimeInHours = 7; // 7 AM
 const dayEndTimeInHours = 19; // 7 PM
 let timerID = null;
 
-class MainView extends PureComponent {
-  constructor(props) {
-    super(props);
+const MainView = (props) => {
+  const getShouldShowOverlay = () => {
+    const now = new Date();
+    const currentTimeInHours = now.getHours();
+
+    const shouldShowOverlay =
+      currentTimeInHours < dayStartTimeInHours ||
+      currentTimeInHours >= dayEndTimeInHours;
+
+    return shouldShowOverlay;
+  };
```

Comment:
The problem here is that:
- When this code is rendered on the server (by Vercel), it might be day, or it might be night.
- The code rendered on the server should _always_ be day-mode (this will improve SEO for our site).
- Then, when the code loads on the client side in the user's browser, if it is night, the screen should appear.

However, this will cause a problem because the user will see a "flash" of content before it is hidden. Do you have any ideas to solve this problem?

Reply:
Oh got it! I might be able to figure it out! let me try few things!

Reply:
It's possible we may need to "fade the modal in" when it loads in the browser, so it's not a flash, and looks intentional... (so maybe it should be a _modal_ actually!)

Reply:
One question:
How can I make sure when code is rendered on the server, it will always be day-mode? 
Should I make the state that checks the day or night to be day in the beginning and add the checker that if `window` is available when it runs the time checker function to update state? `window` will not be available when code is rendered on server, right?

If night mode is modal, we have day-mode content is already available underneath so it won't affect the SEO, right?

Reply:
> How can I make sure when code is rendered on the server, it will always be day-mode?

That's up to you to research and test ;)

> If night mode is modal, we have day-mode content is already available underneath so it won't affect the SEO, right?

No, that's not correct. Google SEO is now very smart, and it will see that the content underneath is hidden. If it is hidden, Google will ignore it.

Plus, it will be a weird experience for users to see the modal, and then it will disappear!